src/VisualStudio/Core/Def/Implementation/ChangeSignature/AddParameterDialogViewModel.cs

```diff
@@ -94,15 +161,87 @@ private void SendFailureNotification(string message)
             _notificationService?.SendNotification(message, severity: NotificationSeverity.Information);
         }
 
-        private bool IsParameterTypeValid(string typeName, Document document)
+        private void SetCurrentTypeTextAndUpdateBindingStatus(string typeName)
+        {
+            VerbatimTypeName = typeName;
+
+            if (typeName.IsNullOrWhiteSpace())
+            {
+                TypeIsEmptyImage = Visibility.Visible;
+                TypeDoesNotParseImage = Visibility.Collapsed;
+                TypeDoesNotBindImage = Visibility.Collapsed;
+                TypeBindsImage = Visibility.Collapsed;
+                TypeBindsDynamicStatus = ServicesVSResources.Please_enter_a_type_name;
+
+                TypeSymbol = null;
+            }
+            else
+            {
+                TypeIsEmptyImage = Visibility.Collapsed;
+
+                var languageService = Document.GetRequiredLanguageService<IChangeSignatureViewModelFactoryService>();
+                TypeSymbol = _semanticModel.GetSpeculativeTypeInfo(InsertPosition, languageService.GetTypeNode(typeName), SpeculativeBindingOption.BindAsTypeOrNamespace).Type;
+
+                var typeParses = IsParameterTypeSyntacticallyValid(typeName);
+                if (!typeParses)
+                {
+                    TypeDoesNotParseImage = Visibility.Visible;
+                    TypeDoesNotBindImage = Visibility.Collapsed;
+                    TypeBindsImage = Visibility.Collapsed;
+                    TypeBindsDynamicStatus = ServicesVSResources.Type_name_has_a_syntax_error;
+                }
+                else
+                {
+                    var parameterTypeBinds = DoesTypeFullyBind(TypeSymbol);
+                    TypeDoesNotParseImage = Visibility.Collapsed;
+
+                    TypeBindsImage = parameterTypeBinds ? Visibility.Visible : Visibility.Collapsed;
+                    TypeDoesNotBindImage = !parameterTypeBinds ? Visibility.Visible : Visibility.Collapsed;
+                    TypeBindsDynamicStatus = parameterTypeBinds
+                        ? ServicesVSResources.Type_name_is_recognized
+                        : ServicesVSResources.Type_name_is_not_recognized;
+                }
+            }
+
+            NotifyPropertyChanged(nameof(TypeBindsDynamicStatus));
+            NotifyPropertyChanged(nameof(TypeBindsImage));
+            NotifyPropertyChanged(nameof(TypeDoesNotBindImage));
+            NotifyPropertyChanged(nameof(TypeDoesNotParseImage));
+            NotifyPropertyChanged(nameof(TypeIsEmptyImage));
```

Comment:
This looks tedious, why do the properties not NotifyPropertyChanged themselves?

Reply:
They don't change, only their Visibility changes. How would they NPC themselves?

Reply:
aren't these values visibility values that get bound? Maybe I'm just confusing some of the names.

Reply:
Talked offline and we could do this either way. Leaving as-is for now.